dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -404,7 +416,7 @@ internal class DatadogRumMonitor(
             }
         )
         writer.write(updatedEvent)
-        updateAndSendView { it.incrementResourceCount() }
+        updateAndSendView(startedEvent.context.viewId) { it.incrementResourceCount() }
```

Comment:
what if we don't keep the allViewEvents there and instead we keep the ViewEvent inside the `unclosedResources` Map and then instead of using the viewId in the `updateAndSendView` we use directly the ViewEvent ? This means that once you consume the resource you release also the ViewEvent for which the resource was started. I am a bit worried about piling up all these events there in that Map without ever releasing it. Even if you say there's not going to be too many views per session this is a Collection that will always grow and never release its memory.

Reply:
The thing is that this will be refactored before we release it, and knowing that I don't want to spend too much time on this, given that we will rewrite it in 2 weeks.